### PR TITLE
Fix secure store module constraint mapping

### LINKED ISSUE #24829: Expo-secure-store incorrectly handles WHEN_UNLOCKED_THIS_DEVICE_ONLY constraint
### Minimal reproducible example

https://snack.expo.dev/@rob.trevethan/smiling-violet-graham-crackers

### Summary

Expo secure store for IOS enables you to set security constraints to manage the secure access to fields saved in the secure store.  These constraints are like always vs when unlocked vs phone passcode and first unlock.

When using the specific constraint `WHEN_UNLOCKED_THIS_DEVICE_ONLY`:
-  this worked fine on expo 48
-  on expo 49  it errors when accessing the secure store on devices that do not have a passcode or biometrics set 
-  it starts to work if you then set a passcode 
-  A device passcode shouldn't be be required for this constraint and Expo 48 was working as expected 
-  the error makes it behave like the constraint `WHEN_PASSCODE_SET_THIS_DEVICE_ONLY`and there is an obvious error in the code that explains why

Code:
- after the Expo 48  the secure store module was [refactored to the new Expo module format](https://github.com/expo/expo/pull/21393) and this was released in expo 49
- You can see a [map of the constraints in the code here](https://github.com/expo/expo/blame/e68cb072a211aaf8ae709d8bfa88acca885b24e8/packages/expo-secure-store/ios/SecureStoreModule.swift#L16) that `WHEN_UNLOCKED_THIS_DEVICE_ONLY` has been mapped to `SecureStoreAccessible.whenPasscodeSetThisDeviceOnly.rawValue` i,e, with passcode which is incorrect.
- It should be mapped to `SecureStoreAccessible.whenUnlockedThisDeviceOnly.rawValue`.  
- The accessible values for `WHEN_UNLOCKED_THIS_DEVICE_ONLY` exists in the mapping for IOS in the [latest version](https://github.com/expo/expo/blob/cf94ec81b80892e360f413fc59f1dcbf245e65a0/packages/expo-secure-store/ios/SecureStoreAccessible.swift#L10) and were also present in the [old module](https://github.com/expo/expo/blob/e68cb072a211aaf8ae709d8bfa88acca885b24e8/ios/versioned/sdk48/EXSecureStore/EXSecureStore/ABI48_0_0EXSecureStore.m#L153C14-L153C58) in expo 48.

So seems this should be handled + work, its just that mapping is incorrect. It appears a minor error was introduced during the refactor of secure store to the new Expo modules.

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 6.2 Ubuntu 23.04 23.04 (Lunar Lobster)
      Shell: 5.2.15 - /bin/bash
    Binaries:
      Node: 16.14.2 - /run/user/1000/fnm_multishells/10533_1697017000792/bin/node
      Yarn: 1.22.18 - /run/user/1000/fnm_multishells/10533_1697017000792/bin/yarn
      npm: 8.5.0 - /run/user/1000/fnm_multishells/10533_1697017000792/bin/npm
    SDKs:
      Android SDK:
        Build Tools: 30.0.2, 30.0.3, 31.0.0, 32.0.0
        System Images: android-29 | Google APIs Intel x86 Atom
    npmPackages:
      @expo/webpack-config: ^19.0.0 => 19.0.0 
      babel-preset-expo: ^9.3.0 => 9.5.1 
      expo: ^49.0.0 => 49.0.7 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.4 => 0.72.4 
    Expo Workflow: bare

### PR DESCRIPTION
**Pull request by mmmguitar**
# Why
Closes #24829
We are experiencing issues with the secure store constraints after updating from expo 48 to 49.  

I raised an issue https://github.com/expo/expo/issues/24829 that details it further but long story short, there is a constraints map that maps constraint -> secure store accessible and there is an incorrect mapping.

# How

Its a simple mapping issue on an enum, I just quickly edited the file.  I have no means to build + test this given the docs for that seem to be mac specific (I'm using linux) and I just don't have time so would hope this fix is simple enough any CI should "just work" or hope someone else in the community that has the dev env already setup could quickly check this out and test it builds.

# Test Plan

I have no test plan given I can't build it locally and I don't really have the time to fathom this on linux.  The change is rather simple and I would hope CI / building is enough if not I would hope someone else in the community that has the dev env already setup could test this works.  It requires testing on a real iphone as well (details on the bug report) and I don't even have one of those...

To test this, I have snack code in the issue I raised that replicates the issue on a real device.  A version of the that app just needs to be run on an IOS device without a passcode/biometrics and the secure store save / get not error.

# Checklist

- [✓] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md)
-- there is no documentation changes needed as this is a bug fix (apart from the changelog)

- [✓] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
-- I made no real changed, just changed the use of an enum value to the right enum value
- [✓] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
-- assuming I copied and pasted the enum value correctly... I really didnt change much

### CODE REVIEW

**Review by alanjhughes — APPROVED** [review]
Thank you for the fix! Can you add a changelog link and I will merge

File: packages/expo-secure-store/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed the 'WHEN_UNLOCKED_THIS_DEVICE_ONLY' constraint being incorrectly mapped wrong secure store accessible ([#24831](https://github.com/expo/expo/pull/24831) by [@mmmguitar](https://github.com/mmmguitar))
```
Comment (mmmguitar) [inline packages/expo-secure-store/CHANGELOG.md:13]:
```suggestion
- Fixed the 'WHEN_UNLOCKED_THIS_DEVICE_ONLY' constraint being incorrectly mapped to wrong secure store accessible ([#24831](https://github.com/expo/expo/pull/24831) by [@mmmguitar](https://github.com/mmmguitar))
```

**Comment by mmmguitar** [comment]
Many thanks as well! I patched the link you suggested and did a small grammar fix on the bug description

**Comment by mmmguitar** [comment]
Just a question if this could be a patch release for expo 49? or would this be have to be out in the next expo release? 

We've had a number of customers affected by this and there's no real fix without changing the security level or advising users to set passcodes on their phones.

**Comment by Bardiamist** [comment]
Would be good to cherry pick into SDK 49 if new release will not be soon

**Comment by FrancescoBonizzi** [comment]
Hi guys! I didn't understand if this commit is released in 49.*.* or if it is still on the repo only!

**Comment by rafavalerio** [comment]
Sorry to re-open this, but I am having some issues on iOS after migrating from SDK48 to SDK50.
I keep getting `KeyChainException: User interaction is not allowed.` after a while, even when I set the items like this:

```node.js
export const storeToken = (name: string, token: string) => {
  return SecureStore.setItem(name, token, {
    keychainAccessible: SecureStore.ALWAYS,
  })
}
```

This seems to be exclusive to iOS, and when generating a production like build. I cannot reproduce the issue on a development-build.

The issue also doesn't happen straight away, but rather after a few hours, most likely when the app is closed / in the background.

Am I missing a config here?

**Comment by josefelbez** [comment]
I’m using expo 50 and got the same issue, im planning to roll back to rn keychain because of this, any updates or a quick fix on this?

u cannot reproduce in dev env by the way only after building it in production, as well sentry only reports the model:
iPhone 15 Pro Max
Arch:arm64

So im guessing is something related to this phone?

**Comment by grondo89** [comment]
Same here, I'm seeing this issue after upgrading to SDK 50, seeing it only in iOS, but with several phone types (iPhone 15 Pro / 15 Pro Max / 14 Pro / 11) . I have keychainAccessible set to "ALWAYS_THIS_DEVICE_ONLY", and also believe this may be related with the app being on background. Is there someone troubleshooting this ongoing issue?

**Comment by justin-medeiros** [comment]
I am also getting this issue when using SDK 50. I can confirm that this issue only happens on iOS and happens after a couple hours of the app being in the background. I have the keychainAccessible set as follows:
```typescript
{
	keychainAccessible: SecureStore.AFTER_FIRST_UNLOCK
}
```
Is there an issue that has been opened for this?

**Comment by federicogomezlara** [comment]
We are unable to access SecureStore values on 50 AND 51, not reproducible on dev clients, whether it's on a simulator or a real device

**Comment by dhcmega** [comment]
Hi! is this solved? thanks

**Pull request merged**